app/models/piece.rb

```diff
@@ -124,13 +124,21 @@ def never_moved?
     updated_at == created_at
   end
 
+  def active_player
+    if self.game.move_counter % 2 == 0
```

Comment:
Since the Game model already has active player, you should probably just "proxy" to it, something like:

```ruby
def active_player
  game.active_player
end
```

There's even a special method in Rails call `delegate` that does this:

```ruby
delegate :active_player, to: :game

# or for multiple methods
delegate :some_method, :some_other_method, :a_third_one, to: :another_method
# so calling instance.some_method would just pass everything to instance.another_method.some_method
```